data-prepper-plugins/type-conversion-processor/src/test/java/org/opensearch/dataprepper/plugins/processor/typeconversion/TypeConversionProcessorTests.java

```diff
@@ -0,0 +1,215 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.plugins.processor.typeconversion;
+
+import org.opensearch.dataprepper.metrics.PluginMetrics;
+import org.opensearch.dataprepper.model.event.Event;
+import org.opensearch.dataprepper.model.event.JacksonEvent;
+import org.opensearch.dataprepper.model.record.Record;
+import org.hamcrest.CoreMatchers;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.regex.PatternSyntaxException;
+
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.CoreMatchers.notNullValue;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.Mockito.lenient;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(MockitoExtension.class)
+public class TypeConversionProcessorTests {
+    @Mock
+    private PluginMetrics pluginMetrics;
+
+    @Mock
+    private TypeConversionProcessorConfig mockConfig;
+
+    private TypeConversionProcessor typeConversionProcessor;
+
+    static Record<Event> buildRecordWithEvent(final Map<String, Object> data) {
+        return new Record<>(JacksonEvent.builder()
+                .withData(data)
+                .withEventType("event")
+                .build());
+    }
+
+    private Record<Event> getMessage(String message, String key, Object value) {
+        final Map<String, Object> testData = new HashMap();
+        testData.put("message", message);
+        testData.put(key, value);
+        return buildRecordWithEvent(testData);
+    }
+
+    private Event executeAndGetProcessedEvent(final Record<Event> record) {
+        final List<Record<Event>> processedRecords = (List<Record<Event>>) typeConversionProcessor.doExecute(Collections.singletonList(record));
+        assertThat(processedRecords.size(), equalTo(1));
+        assertThat(processedRecords.get(0), notNullValue());
+        Event event = processedRecords.get(0).getData();
+        assertThat(event, notNullValue());
+        return event;
+    }
+
+    @Test
+    void testStringToIntegerTypeConversionProcessor() {
+        String testKey = "key1";
```

Comment:
Considering this variable `testKey` is being used in all the test cases, you can make this a static final class variable
```
private static final String TEST_KEY = "key1"
```

Reply:
Additionally, you can call the `mockConfig.getKey()` in a Setup method which is redundant in all the test cases, like this:
```
@BeforeEach
public void setup() {
     lenient().when(mockConfig.getKey()).thenReturn(TEST_KEY);
}
```